infra/lib/api-stack.ts

```diff
@@ -318,53 +321,15 @@ export class APIStack extends Stack {
     });
 
     // token auth for connect sessions
-    const tokenAuthLambda = new lambda_node.NodejsFunction(this, "APITokenAuthLambda", {
-      runtime: lambda.Runtime.NODEJS_16_X,
-      entry: "../api/lambdas/token-auth/index.js",
-      environment: {
-        TOKEN_TABLE_NAME: dynamoDBTokenTable.tableName,
-      },
-    });
-    addErrorAlarmToLambdaFunc(this, tokenAuthLambda, "TokenAuthFunctionAlarm");
-
-    const tokenAuth = new apig.RequestAuthorizer(this, "APITokenAuth", {
-      handler: tokenAuthLambda,
-      identitySources: ["method.request.querystring.state"],
-      // todo: instead of removing caching, investigate explicitly listing
-      //        the permitted methods in the lambda: "Resource: event.methodArn"
-      //
-      // see: https://forum.serverless.com/t/rest-api-with-custom-authorizer-how-are-you-dealing-with-authorization-and-policy-cache/3310
-      resultsCacheTtl: Duration.minutes(0),
-    });
-    tokenAuthLambda.role && dynamoDBTokenTable.grantReadData(tokenAuthLambda.role);
+    const tokenAuth = this.setupTokenAuthLambda(dynamoDBTokenTable);
 
     // setup /token path with token auth
-    const apiTokenResource = api.root.addResource("token");
-    const tokenProxy = new apig.ProxyResource(this, `${id}/token/Proxy`, {
-      parent: apiTokenResource,
-      anyMethod: false,
-    });
-    const integrationToken = new apig.Integration({
-      type: apig.IntegrationType.HTTP_PROXY,
-      options: {
-        connectionType: apig.ConnectionType.VPC_LINK,
-        vpcLink: link,
-        requestParameters: {
-          "integration.request.path.proxy": "method.request.path.proxy",
-          "integration.request.header.api-token": "context.authorizer.api-token",
-          "integration.request.header.cxId": "context.authorizer.cxId",
-          "integration.request.header.userId": "context.authorizer.userId",
-        },
-      },
-      integrationHttpMethod: "ANY",
-      uri: `http://${fargateService.loadBalancer.loadBalancerDnsName}/{proxy}`,
-    });
-    tokenProxy.addMethod("ANY", integrationToken, {
-      requestParameters: {
-        "method.request.path.proxy": true,
-      },
-      authorizer: tokenAuth,
-    });
+    this.setupAPIGWApiTokenResource(id, api, link, tokenAuth, apiServerAddress);
```

Comment:
Just moved this to its own function to move towards a more maintainable main function. Can provide code comparison if needed.